python/cuml/test/test_tsvd.py

```diff
@@ -70,19 +112,40 @@ def test_tsvd_fit_transform(datatype, input_type):
 
 @pytest.mark.parametrize('datatype', [np.float32, np.float64])
 @pytest.mark.parametrize('input_type', ['dataframe', 'ndarray'])
-def test_tsvd_inverse_transform(datatype, input_type):
-    gdf = cudf.DataFrame()
-    gdf['0'] = np.asarray([-1, -2, -3, 1, 2, 3], dtype=datatype)
-    gdf['1'] = np.asarray([-1, -1, -2, 1, 1, 2], dtype=datatype)
+@pytest.mark.parametrize('name', [pytest.param('None', marks=pytest.mark.unit),
+                                  pytest.param('blobs',
+                                               marks=pytest.mark.stress),
+                                  pytest.param('random',
+                                               marks=pytest.mark.quality)])
+def test_tsvd_inverse_transform(datatype, input_type,
+                                name):
+
+    if name == 'blobs':
+        pytest.skip('fails when using blobs dataset')
+        X, y = make_blobs(n_samples=500000,
+                          n_features=1000, random_state=0)
+
+    elif name == 'random':
+        pytest.skip('fails when using random dataset '
+                    'used by sklearn for testing')
+        shape = 5000, 100
+        rng = check_random_state(42)
+        X = rng.randint(-100, 20, np.product(shape)).reshape(shape)
+
+    else:
+        X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]],
+                     dtype=datatype)
+
+    X_pd = pd.DataFrame(
+           {'fea%d' % i: X[0:, i] for i in range(X.shape[1])})
+    X_cudf = cudf.DataFrame.from_pandas(X_pd)
     cutsvd = cuTSVD(n_components=1)
 
     if input_type == 'dataframe':
-        Xcutsvd = cutsvd.fit_transform(gdf)
+        Xcutsvd = cutsvd.fit_transform(X_cudf)
 
     else:
-        X = np.array([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]],
-                     dtype=datatype)
         Xcutsvd = cutsvd.fit_transform(X)
 
     input_gdf = cutsvd.inverse_transform(Xcutsvd)
-    assert array_equal(input_gdf, gdf, 0.4, with_sign=True)
+    assert array_equal(input_gdf, X_cudf, 0.4, with_sign=True)
```

Comment:
Woah, 0.4 is a pretty big tolerance? Ok, this predates your change, so you don't need to change it now, but it's a little worrying...